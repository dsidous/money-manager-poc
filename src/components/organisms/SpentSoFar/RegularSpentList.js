import React from 'react';
import { Text, View, StyleSheet, TouchableOpacity } from 'react-native';
import { StyledText } from '../../atoms';

const styles = StyleSheet.create({
  container: {
    alignItems: 'flex-end',
    borderBottomWidth: 1,
    borderBottomColor: '#ccc',
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingVertical: 20,
  },
  containerLeft: {
    paddingVertical: 5,
  },
  date: {
    fontSize: 18,
  },
  title: {
    fontSize: 23,
  },
  amount: {
    fontSize: 24,
  }
});

const RegularSpentList = ({ data, sliceColor, showHistory }) => data.map((item, i) => (
  <TouchableOpacity
    key={i}
    onPress={() => showHistory(item)}
  >
    <View style={styles.container}>
      <View style={{ flexDirection: 'row' }}>
        <View style={{ alignSelf: 'center' }}>
          <Text style={{ color: sliceColor[i], marginRight: 10 }}>&#9679;</Text>
        </View>

        <View style={styles.containerLeft}>
          <StyledText style={styles.date}>{item.date}</StyledText>
          <StyledText style={styles.title}>{item.title}</StyledText>
        </View>
      </View>
      <View>
        <StyledText style={styles.amount}>{`Â£${item.amount}`}</StyledText>
      </View>
    </View>
  </TouchableOpacity>
));

export default RegularSpentList;
