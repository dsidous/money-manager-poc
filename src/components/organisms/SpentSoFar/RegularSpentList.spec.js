import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import RegularSpentList from './RegularSpentList';

configure({ adapter: new Adapter() });

describe('Regular spent list', () => {
  const data = [{
    category: 'test',
    amount: 1,
  }];
  const sliceColor = [];
  const render = shallow(<RegularSpentList data={data} sliceColor={sliceColor} />);

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });
});
