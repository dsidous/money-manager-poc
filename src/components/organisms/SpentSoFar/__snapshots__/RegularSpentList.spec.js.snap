// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Regular spent list renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RegularSpentList
    data={
      Array [
        Object {
          "amount": 1,
          "category": "test",
        },
      ]
    }
    sliceColor={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": "0",
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Component
          style={
            Object {
              "flexDirection": "row",
            }
          }
        >
          <Component
            style={
              Object {
                "alignSelf": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "color": undefined,
                  "marginRight": 10,
                }
              }
            >
              ●
            </Text>
          </Component>
          <Component
            style={
              Object {
                "paddingVertical": 5,
              }
            }
          >
            <StyledText
              style={
                Object {
                  "fontSize": 18,
                }
              }
            />
            <StyledText
              style={
                Object {
                  "fontSize": 23,
                }
              }
            />
          </Component>
        </Component>,
        <Component>
          <StyledText
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            £1
          </StyledText>
        </Component>,
      ],
      "style": Object {
        "alignItems": "flex-end",
        "borderBottomColor": "#ccc",
        "borderBottomWidth": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "paddingVertical": 20,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Component
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": undefined,
                    "marginRight": 10,
                  }
                }
              >
                ●
              </Text>
            </Component>,
            <Component
              style={
                Object {
                  "paddingVertical": 5,
                }
              }
            >
              <StyledText
                style={
                  Object {
                    "fontSize": 18,
                  }
                }
              />
              <StyledText
                style={
                  Object {
                    "fontSize": 23,
                  }
                }
              />
            </Component>,
          ],
          "style": Object {
            "flexDirection": "row",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": undefined,
                    "marginRight": 10,
                  }
                }
              >
                ●
              </Text>,
              "style": Object {
                "alignSelf": "center",
              },
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "accessible": true,
                "allowFontScaling": true,
                "children": "●",
                "ellipsizeMode": "tail",
                "style": Object {
                  "color": undefined,
                  "marginRight": 10,
                },
              },
              "ref": null,
              "rendered": "●",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <StyledText
                  style={
                    Object {
                      "fontSize": 18,
                    }
                  }
                />,
                <StyledText
                  style={
                    Object {
                      "fontSize": 23,
                    }
                  }
                />,
              ],
              "style": Object {
                "paddingVertical": 5,
              },
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": undefined,
                  "style": Object {
                    "fontSize": 18,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": undefined,
                  "style": Object {
                    "fontSize": 23,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <StyledText
            style={
              Object {
                "fontSize": 24,
              }
            }
          >
            £1
          </StyledText>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "£1",
            "style": Object {
              "fontSize": 24,
            },
          },
          "ref": null,
          "rendered": "£1",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": "0",
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Component
            style={
              Object {
                "flexDirection": "row",
              }
            }
          >
            <Component
              style={
                Object {
                  "alignSelf": "center",
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "color": undefined,
                    "marginRight": 10,
                  }
                }
              >
                ●
              </Text>
            </Component>
            <Component
              style={
                Object {
                  "paddingVertical": 5,
                }
              }
            >
              <StyledText
                style={
                  Object {
                    "fontSize": 18,
                  }
                }
              />
              <StyledText
                style={
                  Object {
                    "fontSize": 23,
                  }
                }
              />
            </Component>
          </Component>,
          <Component>
            <StyledText
              style={
                Object {
                  "fontSize": 24,
                }
              }
            >
              £1
            </StyledText>
          </Component>,
        ],
        "style": Object {
          "alignItems": "flex-end",
          "borderBottomColor": "#ccc",
          "borderBottomWidth": 1,
          "flexDirection": "row",
          "justifyContent": "space-between",
          "paddingVertical": 20,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <Component
                style={
                  Object {
                    "alignSelf": "center",
                  }
                }
              >
                <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": undefined,
                      "marginRight": 10,
                    }
                  }
                >
                  ●
                </Text>
              </Component>,
              <Component
                style={
                  Object {
                    "paddingVertical": 5,
                  }
                }
              >
                <StyledText
                  style={
                    Object {
                      "fontSize": 18,
                    }
                  }
                />
                <StyledText
                  style={
                    Object {
                      "fontSize": 23,
                    }
                  }
                />
              </Component>,
            ],
            "style": Object {
              "flexDirection": "row",
            },
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                    Object {
                      "color": undefined,
                      "marginRight": 10,
                    }
                  }
                >
                  ●
                </Text>,
                "style": Object {
                  "alignSelf": "center",
                },
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "accessible": true,
                  "allowFontScaling": true,
                  "children": "●",
                  "ellipsizeMode": "tail",
                  "style": Object {
                    "color": undefined,
                    "marginRight": 10,
                  },
                },
                "ref": null,
                "rendered": "●",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <StyledText
                    style={
                      Object {
                        "fontSize": 18,
                      }
                    }
                  />,
                  <StyledText
                    style={
                      Object {
                        "fontSize": 23,
                      }
                    }
                  />,
                ],
                "style": Object {
                  "paddingVertical": 5,
                },
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": undefined,
                    "style": Object {
                      "fontSize": 18,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "children": undefined,
                    "style": Object {
                      "fontSize": 23,
                    },
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <StyledText
              style={
                Object {
                  "fontSize": 24,
                }
              }
            >
              £1
            </StyledText>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "£1",
              "style": Object {
                "fontSize": 24,
              },
            },
            "ref": null,
            "rendered": "£1",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
